crates/adapters/src/catalog.rs

```diff
@@ -227,7 +227,11 @@ impl Catalog {
         let circuit = stream.circuit();
 
         // Create handle for the stream itself.
-        let delta_handle = stream.output();
+        // Make sure we only output unit weights.
+        let delta_handle = stream.distinct().output();
```

Comment:
If it's a zset it should not be distinct. Some queries output multisets

Reply:
Good point.  The function should be called `add_input_set`.

> Some queries output multisets

Do you have examples of such queries?  I need to understand what exactly we promise about output multiplicities.

Reply:
Select col from table

Reply:
Well, we could say that we interpret such rules using set semantics.  If we want to maintain the multiset semantics then we'd need to figure out what to do about operations like joins, which multiply weights. What does SQL say about those?

Reply:
I am not sure what we are arguing about. The compiler correctly handles sql semantics. Some sql queries return multisets. So outputs can be multisets. The compiler will insert distincts where the sql semantics requires it. Plan rewrites may generate joins that also output multisets. Dbsp should give the compiler the building blocks it needs to express sql semantics.

Reply:
I don't think we're arguing.  If the compiler correctly handles multiset semantics, then we shouldn't distinct the outputs, so that part of the PR is wrong.  But now I need to figure out how to handle non-unit weights.  Previously, we output numeric weights, which is correct, but seems confusing to users.  Alternatively, we can represent weight `n` by outputting `n` inserts (and `-n` by outputting `n` deletes), but that only works for relatively small `n`'s, while there's nothing preventing users from writing queries that produce very large `n`'s (we even have an issue on dealing with overflows).

Another question is whether we allow multiset inputs.  I suppose we should, for tables without primary/unique keys, but we no longer do, because we apply `upsert` to all inputs.  `upsert` is mostly needed to prevent negative weights, so we can add a new operator that allows multisets with arbitrary positive multiplicities, but prunes updates that would introduce negative weights.

Reply:
Yes, tables are multisets and there are quite a few tests treating them that way. More subtle is the fact that the column types are în fact implicit casts for the inserted values, so in sql even distinct inserted rows can generate identical inserted values. But the compiler doesn't handle that, it only allows properly typed values. The serde will do the implicit cast.

Reply:
But the compiler may need to help the serde. I think we have at least one issue about it

Reply:
My solution for now is to output `N` `insert`/`delete` records for each output update with weight `N`. If `N` is very large (>1M), I will raise an error. We can add support for other JSON encodings, including the encoding with explicit weights.